plugins/modules/vmware_guest.py

```diff
@@ -2477,7 +2477,7 @@ def sanitize_disk_parameters(self, vm_obj):
         """
         controllers = []
         for disk_spec in self.params.get('disk'):
-            if not disk_spec['controller_type'] or not disk_spec['controller_number'] or not disk_spec['unit_number']:
```

Comment:
Could you please add an integration test for this change? REST LGTM

Reply:
@Akasurde I think there already exists a test case for `controller_number` / `unit_number` 0:

https://github.com/ansible-collections/community.vmware/blob/f913056612c776b6527aa3c373fe7cb7358d2762/tests/integration/targets/vmware_guest/tasks/multiple_disk_controllers_d1_c1_f0.yml#L21-L30

But I don't understand why this didn't fail in the CI :-/